common/lazy-pages/src/lib.rs

```diff
@@ -105,27 +127,38 @@ pub fn update_lazy_pages_and_protect_again(
             new_mem_addr
         );
 
-        // Cannot panic, unless OS allocates wasm mem buffer
-        // in not aligned by native page addr.
-        gear_ri::set_wasm_mem_begin_addr(new_mem_addr);
-    }
+        Some(new_mem_addr)
+    } else {
+        None
+    };
 
     let new_mem_size = mem.size();
-    if new_mem_size > old_mem_size {
-        gear_ri::set_wasm_mem_size(new_mem_size.into());
+    let changed_size = match new_mem_size > old_mem_size {
+        true => Some(new_mem_size.raw()),
+        false => None,
+    };
+
+    if !matches!((changed_addr, changed_size), (None, None)) {
+        gear_ri::change_wasm_memory_addr_and_size(changed_addr, changed_size)
     }
 
     mprotect_lazy_pages(mem, true);
 }
 
 /// Returns list of released pages numbers.
-pub fn get_released_pages() -> Vec<GearPage> {
-    gear_ri::get_released_pages()
+pub fn get_write_accessed_pages() -> Vec<GearPage> {
+    gear_ri::write_accessed_pages()
+        .iter()
+        .map(|p| {
+            GearPage::new(*p)
```

Comment:
We can consume the vector.
```suggestion
        .into_iter()
        .map(|p| {
            GearPage::new(p)
```